### PR TITLE
Added unit tests

### PR DESCRIPTION
**Pull request by ifitzsimmons**

### CODE REVIEW

File: Demographic_Study/demographic_test.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import unittest
+import numpy as np
+import pandas as pd
+import find_similar_states as fss
+
+def test_filter_by_race():
+    test_frame = pd.DataFrame({
+        'White': [], 'Black': [], 'Hispanic': [], 'Asian': [],
```
Comment (johnhoman) [inline Demographic_Study/demographic_test.py]: data?

File: Demographic_Study/demographic_test.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    result = fss.filter_by_race(test_frame)
+    pd.testing.assert_frame_equal(result.sort_index(axis=1), expected.sort_index(axis=1))
+
+def test_cos_sim():
+    array_1 = np.array([1, 0, 0])
+    array_2 = np.array([1, 2, 2])
+
+    expected = 1.0/3
+    result = fss.get_cos_sim(array_1, array_2)
+    assert result == expected
```
Comment (johnhoman) [inline Demographic_Study/demographic_test.py]: floating point comparison. 
```python
assert result == pytest.approx(expected)
```

File: Demographic_Study/demographic_test.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+def test_get_state_similarities():
+    test_frame = pd.DataFrame({
+        'White': [1,1,10,4], 'Black': [1,1,20,4], 'Hispanic': [0,0,50,1]
+    })
+    indeces = ['A', 'B', 'C', 'D']
```
Comment (johnhoman) [inline Demographic_Study/demographic_test.py]: spelling

File: Demographic_Study/demographic_test.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        ('A', 'D'),
+        ('A', 'C'),
+        ('A', 'B'),
+    ]
+
+    most, least = fss.get_sim_info(test)
+    assert most_expected == most
+    assert least_expected == least
+
+if __name__ == '__main__':
```
Comment (johnhoman) [inline Demographic_Study/demographic_test.py]: so this isn't correct, it should be something at the root like a makefile
```Makefile
test:
    @find . -type d -name __pycache__ | xargs rm -rf
    @find . -type d -name "*.pyc" | xargs rm -rf
    python -m pytest tests
```

File: Demographic_Study/tests/test_find_similar_states.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        ('B', 'D', fss.get_cos_sim(array_dict['B'], array_dict['D'])),
+        ('C', 'D', fss.get_cos_sim(array_dict['C'], array_dict['D'])),
+        ('A', 'C', fss.get_cos_sim(array_dict['A'], array_dict['C'])),
+        ('B', 'C', fss.get_cos_sim(array_dict['B'], array_dict['C']))
+    ]
+
+    result = fss.get_state_similarities(test_frame)
+    assert result == expected
+
+if __name__ == '__main__':
```
Comment (johnhoman) [inline Demographic_Study/tests/test_find_similar_states.py]: same as other file

File: Demographic_Study/tests/test_find_similar_states.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
+import unittest
```
Comment (johnhoman) [inline Demographic_Study/tests/test_find_similar_states.py]: you don't need unittest here. Use pytest

File: Demographic_Study/demographic_test.py (hunk 4 above)
Comment (ifitzsimmons) [reply Demographic_Study/demographic_test.py]: do I add the first part to the file and the second part is command line?

Comment (johnhoman) [reply Demographic_Study/demographic_test.py]: yes, the first part is the `Makefile`, the second part is from a bash prompt

File: Demographic_Study/tests/test_find_similar_states.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    array_1 = np.array([1, 0, 0])
+    array_2 = np.array([1, 2, 2])
+
+    expected = 1.0/3
+    result = fss.get_cos_sim(array_1, array_2)
+    assert result == expected
+
+def test_get_state_similarities():
+    test_frame = pd.DataFrame({
+        'White': [1,1,10,4], 'Black': [1,1,20,4], 'Hispanic': [0,0,50,1]
```
Comment (johnhoman) [inline Demographic_Study/tests/test_find_similar_states.py]: always a space after commas

File: Demographic_Study/find_similar_states.py [hunk 8]
```diff
@@ -46,6 +47,26 @@ def filter_by_race(df):
     
     return races_df
 
+def get_cos_sim(array_1, array_2):
+    """Calculate cosine similarity of two numpy arrays
+    
+    Parameters
+    ----------
+    array_1, array_2: np.array
```
Comment (johnhoman) [inline Demographic_Study/find_similar_states.py]: separate these and put the expected/supported shapes of the arrays so it's clear
```python
def get_cos_sim(array_1, array_2):
    """
    array_1:  np.ndarray, shape [n_samples, m_columns]
    array_2:  np.ndarray, shape [n_samples, m_columns]
    """
```

File: Demographic_Study/find_similar_states.py [hunk 9]
```diff
@@ -110,10 +131,12 @@ def get_sim_info(similarities, top):
     least_similar = [(item[0], item[1]) for item in reversed(similarities[-top:])]
     return most_similar, least_similar
 
+
 if __name__ == "__main__":
     top_results = _parse_args().parse_args().top_results
     path = 'data/demographics.csv'
 
+    get_sim_info
```
Comment (johnhoman) [inline Demographic_Study/find_similar_states.py]: What's the deal with this... looks like you forgot to call it
```python
get_sim_info(pd.read_csv(path), top=top_results)
```

Comment (johnhoman) [inline Demographic_Study/find_similar_states.py]: why's this here?

**Pull request merged**